### ISSUE TITLE
Any chance of some code?

### ISSUE THREAD

**Issue by chrissype** [AdenosHermes/NeurIPS_2020_Snake#1]
Would love to see the experiment reprod code as the results looks very promising!

**Comment by liu-ziyin** [comment]
Thanks for your interest in our work! We are on our way cleaning our code! It should be ready within a week or two. We will let you know once we make an update.

In the mean-while, let me comment that Edward Dixon had made an implementation of our method a while ago. You can check it here:
https://github.com/EdwardDixon/snake

Ziyin

**Comment by chrissype** [comment]
@AdenosHermes My main question was whether for the the regression experiments, the input feature for each time step was just the (possibly normalized) timestamp, i.e. a monotonically increasing linear feature - I've tried with this on my own dataset and can't repro the nice periodic learned function results.

**Comment by liu-ziyin** [comment]
Thanks for your question!

Let me clarify a few things. (1) For all the regression tasks, what we do is not to train on a single time step, but on the whole data set, by batch gradient descent for every training step (or by minibatch GD). Please try if this works.

If it does not, then let me give a little more comment.
(2) The proposed method is quite sensitive to the choice of hyperparameter $a$ at this moment, so choosing a good $a$ may be crucial for many applications. The good way to choose $a$ is through validation on a small held-out dataset. 

So this point suggests that there is a lot of room to improve our current method (to make it less sensitive to $a$ or to remove the tuning part altogether, say) in the possible future works

**Comment by chrissype** [comment]
So that was also the setting I was using, but my question was, given the classic regression setting of predicting y from a vector of features X, what exactly was X in your experiments? Was it the unix timestamp? A one-hot vector encoding of day of week/month/year etc.?

**Comment by liu-ziyin** [comment]
I see what you mean! Yes, the input is just the time itself, a one-dimensional scalar.

**Comment by chrissype** [comment]
Ah okay, very interesting. Thank you!

**Comment by liu-ziyin** [comment]
Hey Chris! Let me comment that we have updated some of our codes, and, therefore, I will close this issue!
